addons/hr_recruitment/models/hr_candidate.py

```diff
@@ -109,7 +109,7 @@ def _inverse_partner_email(self):
             if not candidate.partner_id:
                 if not candidate.partner_name:
                     raise UserError(_('You must define a Contact Name for this candidate.'))
-                candidate.partner_id = self.env['res.partner'].with_context(default_lang=self.env.lang).find_or_create(candidate.email_from)
+                candidate.partner_id = self.env['res.partner'].with_context(default_lang=self.env.lang).find_or_create(f"{candidate.partner_name} <{candidate.email_from}>")
```

Comment:
```suggestion
                candidate.partner_id = self.env['res.partner'].with_context(default_lang=self.env.lang).find_or_create(candidate.email_from)
```
why not just giving the email since the method expects to have an email in parameter? 🤔 

https://github.com/odoo/odoo/blob/6b820eb6fc6f782ba6a83d605d87b4a1dd2a87be/odoo/addons/base/models/res_partner.py#L916